### ISSUE TITLE
SEB is running into "SEB GESPERRT" mode seemingly randomly

### ISSUE THREAD

**Issue by time4breakfast** [SafeExamBrowser/seb-win-refactoring#914]
**Describe the Bug**
We use Moodle tests with safe exam browser and have the SEB installed on a number of notebooks and wanted to show it to students in order to prepare them for the exam. Every now and then the SEB shows the red screen with either one of two messages:
- SEB gesperrt (verbotene Barrierefreiheit-Konfiguration)
- SEB gesperrt (letzte Sitzung nicht ordnungsgemäß beendet) 
Sometimes it is possible to enter the admin password and get out of the situation, sometimes this is not possible and the only way to get out is to do a hard reset (switch the notebook off).

Additionally, sometimes the user gets shown a message saying, that chrome.exe needs to be shut down for SEB to work properly, but it cannot find any chrome.exe tasks to end so it is kind of in an endless loop.

I attached the logs files of five of the affected machines.

From what I could determine so far it seems that the Windows update process (wuauserv) is causing this conflicting behaviour. I found that all notebooks had been switched off for a few days and Windows was trying to get information on updates as well as updates in the background while the SEB was running. In two cases it also seemed that some GPU related processes crashed and couldn't properly recover. This would explain the randomness of the situation.
I was also able to detect in the logs that the password was entered correctly but this didn't resolve the red screen. The only way was then to shut the notebook off completely and restart it.

**Steps to Reproduce**
1. Login to course
2. Click on "start safe exam browser"
3. Login to course again
4. Click on "start course"
5. At some point (unrelated to the question (i.e. doesn't matter if its the first or fourth or ...), unrelated to the time or student) the view switches from the test to a completely red screen saying SEB gesperrt (see screenshots attached)

**Expected Behavior**
The Moodle test to run through without any random red screens.

**Screenshots**
![SEB_error](https://github.com/SafeExamBrowser/seb-win-refactoring/assets/20875722/8fdd7f6b-ecea-4561-9e79-bcb6041350d2)
![PXL_20240612_082148133](https://github.com/SafeExamBrowser/seb-win-refactoring/assets/20875722/e9f7b850-e473-4056-b017-b9a54cd6034b)
![SEB_error_1](https://github.com/SafeExamBrowser/seb-win-refactoring/assets/20875722/41ba2eeb-7cd6-40bc-a487-847ffe5f975a)

[Logs_1.zip](https://github.com/user-attachments/files/15821864/Logs_1.zip)
[Logs_2.zip](https://github.com/user-attachments/files/15821882/Logs_2.zip)
[Logs_3.zip](https://github.com/user-attachments/files/15821893/Logs_3.zip)
[Logs_4.zip](https://github.com/user-attachments/files/15821897/Logs_4.zip)
[Logs_5.zip](https://github.com/user-attachments/files/15821901/Logs_5.zip)

**Version Information**
 - OS: Win11
 - SEB 3.7.0, Build 3.7.0.682

**Additional Context**

**Comment by dbuechel** [comment]
Thanks for the report. The first issue is actually a bug as the log reveals, particularly when reconfiguring from a session which was not using the service component to one which does:

Runtime:

```
2024-06-13 14:26:46.930 [07] - INFO: Initializing service...
2024-06-13 14:26:46.930 [07] - INFO: The service will be ignored for the next session.
...
2024-06-13 14:29:55.314 [14] - DEBUG: [RuntimeHost] Received message 'ReconfigurationMessage', sending response 'SimpleResponse -> Acknowledged'.
...
2024-06-13 14:29:55.439 [07] - INFO: Initializing service...
2024-06-13 14:29:55.440 [07] - DEBUG: [ServiceProxy] Trying to connect to endpoint 'net.pipe://localhost/safeexambrowser/service'...
2024-06-13 14:29:55.441 [07] - DEBUG: [ServiceProxy] Communication channel is opening...
2024-06-13 14:29:55.466 [07] - DEBUG: [ServiceProxy] Communication channel has been opened.
2024-06-13 14:29:55.481 [07] - DEBUG: [ServiceProxy] Connection was established.
2024-06-13 14:29:55.481 [07] - INFO: The service is mandatory and connected.
2024-06-13 14:29:55.483 [07] - INFO: Starting new service session...
2024-06-13 14:29:55.484 [07] - DEBUG: [ServiceProxy] Sending message 'SessionStartMessage'...
2024-06-13 14:29:55.670 [07] - DEBUG: [ServiceProxy] Received response 'SimpleResponse -> Acknowledged' for message 'SessionStartMessage'.
2024-06-13 14:29:55.671 [07] - DEBUG: [ServiceProxy] Service acknowledged session start.
2024-06-13 14:29:56.893 [07] - INFO: Successfully started new service session.
```

Client:

```
2024-06-13 14:29:55.309 [20] - DEBUG: [RuntimeProxy] Sending message 'ReconfigurationMessage'...
2024-06-13 14:29:55.318 [20] - DEBUG: [RuntimeProxy] Received response 'SimpleResponse -> Acknowledged' for message 'ReconfigurationMessage'.
2024-06-13 14:29:55.318 [20] - DEBUG: [RuntimeProxy] Runtime acknowledged reconfiguration request.
2024-06-13 14:29:55.319 [20] - INFO: Sent reconfiguration request for 'C:\Users\Admin\AppData\Local\SafeExamBrowser\Temp\config.seb' to the runtime.
2024-06-13 14:29:55.930 [30] - DEBUG: [Registry] Value 'Debugger' from registry key 'HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\Utilman.exe' has changed from '' to 'SebDummy.exe'!
2024-06-13 14:29:55.932 [30] - WARNING: The ease of access registry value 'HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\Utilman.exe\Debugger' has changed from '' to 'SebDummy.exe'! Attempting to show lock screen...
2024-06-13 14:29:55.938 [30] - INFO: Showing lock screen...
2024-06-13 14:29:56.006 [19] - DEBUG: [ApplicationMonitor] Window has changed from 'SplashScreen' (67110) to '' (132904).
2024-06-13 14:29:56.898 [06] - DEBUG: [ClientHost] Received message 'SimpleMessage -> Shutdown', sending response 'SimpleResponse -> Acknowledged'.
```

I shall fix the issue for the upcoming release version 3.8.0.

With respect to the second lock screen: This is intended behavior, namely if a secure session (a session protected with a quit password) was not terminated normally (by terminating SEB e.g. via quit button or a quit link), then this lock screen will be activated when SEB is restarted with the same configuration or Start URL in order to prevent students from cheating by exiting and re-entering SEB at will.

**Comment by time4breakfast** [comment]
Thank you for your feedback. Looking forward to the new version.

**Comment by dbuechel** [comment]
I'll report back once the issue has been fixed in a beta version of 3.8.0.

**Comment by dbuechel** [comment]
The issue should be resolved and can be tested in the latest development build of version 3.8.0: https://sebdev.ethz.ch/api/buildjobs/nun1i0gqe87nax64/artifacts/SEB_3.8.0.719_SetupBundle.exe.
